commander/src/commands/transaction/create/multisignature.ts

```diff
@@ -0,0 +1,137 @@
+import { getPassphraseFromPrompt } from '../../../utils/reader';
+
+export default class MultisignatureCommand extends BaseCommand {
+	static args = [
+		{
+			name: 'nonce',
+			required: true,
+			description: 'Nonce of the transaction.',
+		},
+		{
+			name: 'fee',
+			required: true,
+			description: 'Transaction fee in LSK.',
+		},
+	];
+
+	static flags = {
+		...BaseCommand.flags,
+		networkIdentifier: flagParser.string(commonFlags.networkIdentifier),
+		passphrase: flagParser.string(commonFlags.passphrase),
+		'number-of-signatures': flagParser.integer({
+			description:
+				'Number of signatures required to validate transactions from that account',
+		}),
+		'mandatory-key': flagParser.string({
+			multiple: true,
+			description: 'Mandatory public key for multisignature account.',
+		}),
+		'optional-key': flagParser.string({
+			multiple: true,
+			description: 'Optional public key for multisignature account.',
+		}),
+		'member-passphrase': flagParser.string({
+			multiple: true,
+			description:
+				'Number of signatures required to validate transactions from that account. Its a less secure way to sign transaction. See command description for details. ',
+		}),
+	};
+
+	static description = `
+	Creates a transaction which will register the account as a multisignature account if broadcast to the network, using the following arguments:
+
+		1. Mandatory keys to verify signatures for the multisignature account. Signatures for these kes must exists to verify every transaction from that account.
+		2. Optional keys to verify signatures for the multisignature account. Signatures for these kes can or can't exists to verify transaction from that account.
+		3. Number of signatures required for a transaction from the account to be valid. These value must be minimum of count of mandatory keys.
+		4. Member passphrases for every mandatory and optional key provided.
+
+	Use of '--member-passphrase' flag is not encourage for better security practices. This will expose the passphrase to command line and may end up logging into system logs. Use 'transaction:sign' command to sign with member passphrases.
+	`;
+
+	static examples = [
+		'transaction:create:multisignature 1 100 --mandatory-key="xxx" ' +
+			'--mandatory-key="yyy" --optional-key="yyy" --optional-key="yyy" ' +
+			'--number-of-signatures=4 --passphrase="****" ' +
+			'--member-passphrase="****" --member-passphrase="****" ',
+	];
+
+	async run(): Promise<void> {
+		const {
+			args: { fee: feeSource, nonce },
+			flags: {
+				networkIdentifier: networkIdentifierSource,
+				passphrase: passphraseSource,
+				'member-passphrase': memberPassphrasesSource,
+				'optional-key': optionalKeysSource,
+				'mandatory-key': mandatoryKeysSource,
+				'number-of-signatures': numberOfSignaturesSource,
+			},
+		} = this.parse(MultisignatureCommand);
+
+		const numberOfSignatures = numberOfSignaturesSource || 0;
+		const mandatoryKeys = mandatoryKeysSource || [];
```

Comment:
this is `[]` if it doesn't exist and same for below with `multiple: true`?

Reply:
No it appears `undefined` if non is provided.